app/controllers/admin/listings_controller.rb

```diff
@@ -49,8 +49,9 @@ def listing_params
     end
 
     def handle_publish_status
-      unpublish_listing if listing_params[:published] == "0"
-      publish_listing if listing_params[:published] == "1"
+      @listing = Listing.find(params[:id])
```

Comment:
Thanks for the contribution.  We already have the `@listing` from the `create` method.  We shouldn't need to re-find the object.

Were you able to run this in the application to verify behavior?